### PR TITLE
Add md files

### LINKED ISSUE #8: Add documentation
- [x] reqs with links to specs
- [x] specs with links to test
- [x] edit links to tests based on latest updates
- [x] corcutoff needs updating in rox comments and rmd
image
- [x] @zhiiiyang I also noticed a second .gitignore file in the vignette folder, does this need deletion?

### PR DESCRIPTION
**Pull request by NicoleEO**
Adds reqs and specs with links
Resolves #8

### CODE REVIEW

**Comment by NicoleEO** [comment]
@zhiiiyang I resolved the conflict

File: specs.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+2.The function outputs para, a matrix of estimated parameters in iter=2. This matrix has features_all in the columns and parameters(regression coefficients, threshold, r) in the rows.  Both threshold and r are positive.   
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-fitNBthDE.R#L114
+3.The function outputs sizefact, a vector of size factors, when sizefactrec=FALSE, sizefact is the same as sizefact_start.  
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-fitNBthDE.R#L128
+
+
+#### Specs for fitNBthmDE:
+1. The function outputs para, a matrix of estimated parameters. This matrix has features_all in the columns and parameters(regression coefficients, threshold, r) in the rows. Both threshold and r are positive.  
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-fitNBthmDE.R#L89
+
```
Comment (zhiiiyang) [inline specs.md:171]: Please add L216

File: specs.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+#### Specs for coefNBth:
+1. when fullpara=TRUE, the output parameters should be regression coefficients, threshold and r in a list. Both threshold and r are positive.  
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-coefNBth.R#L22
```
Comment (zhiiiyang) [inline specs.md:175]: Please add L57

File: specs.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+2. when fullpara=FALSE, the output parameters should be regression coefficients only in a list  
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-coefNBth.R#L30
```
Comment (zhiiiyang) [inline specs.md:177]: Please add L65

File: specs.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+
+#### Specs for contrastNBth:
+1. The function takes in a DE model as an input from fitNBthDE or fitNBthmDE  
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-contrastNBth.R#L19
```
Comment (zhiiiyang) [inline specs.md:182]: Please add L59

File: specs.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+2. The user input test:statistical test, choose from c("two-sided", ">", "<")   
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-contrastNBth.R#L20
```
Comment (zhiiiyang) [inline specs.md:184]: Please add L60

File: specs.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+3. In the output list, the p values of '>' and '<' for the same variable/feature should add up to 1   
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-contrastNBth.R#L30
```
Comment (zhiiiyang) [inline specs.md:186]: Please add L70

File: specs.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+
+#### Specs for DENBth:
+1. In the output data.frame, the DE table should not have NA if NAto1=TRUE   
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-DENBth.R#L54
```
Comment (zhiiiyang) [inline specs.md:191]: Please add L22

File: specs.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+2. In the output data.frame, the DE table has a column padj when adjp=TRUE   
+test: https://github.com/Nanostring-Biostats/GeoDiff/blob/eef13efc1636fd86e3b833c26967d3fb2c350396/tests/testthat/test-DENBth.R#L57
```
Comment (zhiiiyang) [inline specs.md:193]: Please add L25

**Review by zhiiiyang — CHANGES_REQUESTED** [review]

**Comment by zhiiiyang** [comment]
@NicoleEO, I added a few comments about pinpointing the unit tests for both WTA/CTA or different cases. Let me know if you have any questions.

**Review by zhiiiyang — APPROVED** [review]

**Pull request merged**